src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -47,7 +49,6 @@ internal sealed partial class MiscellaneousFilesWorkspace : Workspace, IVsRunnin
         private readonly Dictionary<uint, (ProjectId projectId, SourceTextContainer textContainer)> _docCookiesToProjectIdAndContainer = new Dictionary<uint, (ProjectId, SourceTextContainer)>();
```

Comment:
I must admit: I'm curious what this would look like if we replaced this dictionary operating on cookies with file names. Would this then let us toss all the cookie passing in common helper? That'd be great then when we replace this with an editor API that wouldn't be talking about cookies anyways.

Reply:
And maybe this isn't a strict requirement _now_, but if you squint a bit you could imagine this isn't a RunningDocumentTableEventsTracker, but just an generic abstraction over "give me the open files and let me know when that changes" -- we might be able to get rid of cookies entirely. That might be handy if we want to use the misc files workspace somewhere else since once this isn't touching the RDT directly it's pretty light on other VS tie-ins.